internal/bminventory/inventory.go

```diff
@@ -675,17 +620,18 @@ func (b *bareMetalInventory) InstallCluster(ctx context.Context, params installe
 	if err = b.refreshAllHosts(ctx, &cluster); err != nil {
 		return common.GenerateErrorResponder(err)
 	}
+	if _, err = b.clusterApi.RefreshStatus(ctx, &cluster, b.db); err != nil {
+		return common.GenerateErrorResponder(err)
+	}
 
 	// Reload again after refresh
 	if err = b.db.Preload("Hosts", "status <> ?", host.HostStatusDisabled).First(&cluster, "id = ?", params.ClusterID).Error; err != nil {
 		return common.NewApiError(http.StatusNotFound, err)
 	}
-	if err = b.verifyClusterNetworkConfig(ctx, &cluster); err != nil {
-		return common.GenerateErrorResponder(err)
-	}
-
-	if err = b.validateAllHostsCanBeInstalled(&cluster); err != nil {
-		return common.GenerateErrorResponder(err)
+	// Verify cluster status
+	if swag.StringValue(cluster.Status) != models.ClusterStatusReady {
```

Comment:
Why does the API need to know about cluster status? 
suggesting to add something like this to cluster API
```
func (m *Manager) IsReadyForInstallation(c *common.Cluster) (bool, string) {
	if swag.StringValue(c.Status) != models.ClusterStatusReady {
		return false, swag.StringValue(c.StatusInfo)
	}
	return true, ""
}
```
and then replace this line with:
```
	if ok, reason := b.clusterApi.IsReadyForInstallation(&cluster); !ok {
		return common.NewApiError(http.StatusConflict,
			errors.Errorf("Cluster is not ready for installation, %s", reason))
	}
```

Reply:
fixed